packages/ui/src/components/BottomSheet/BottomSheet.tsx

```diff
@@ -0,0 +1,169 @@
+  const backdropRef = useRef<HTMLDivElement>(null);
+  const startYRef = useRef(0);
+
+  useEffect(() => {
+    if (isOpen) {
+      setIsVisible(true);
+      setDragY(0);
+      document.body.style.overflow = 'hidden';
+    } else {
+      document.body.style.overflow = 'unset';
+      const timer = setTimeout(() => {
+        setIsVisible(false);
+        setDragY(0);
+      }, 300);
+      return () => clearTimeout(timer);
+    }
+
+    return () => {
+      document.body.style.overflow = 'unset';
+    };
+  }, [isOpen]);
+
+  const handleBackdropClick = (e: React.MouseEvent) => {
+    if (enableBackdropClose && e.target === backdropRef.current) {
+      onClose();
+    }
+  };
+
+  const handleDragEnd = useCallback(() => {
+    if (!isDragging) return;
+
+    setIsDragging(false);
+    if (dragY > 100) {
+      onClose();
+    } else {
+      setDragY(0);
+    }
+  }, [isDragging, dragY, onClose]);
+
+  const handleMouseDown = useCallback((e: React.MouseEvent) => {
+    if ((e.target as HTMLElement).closest('button')) return;
+
+    e.preventDefault();
+    setIsDragging(true);
+    startYRef.current = e.clientY;
+    setDragY(0);
+  }, []);
+
+  const handleTouchStart = useCallback((e: React.TouchEvent) => {
+    setIsDragging(true);
+    startYRef.current = e.touches[0].clientY;
+    setDragY(0);
+  }, []);
+
+  const handleTouchMove = useCallback(
+    (e: React.TouchEvent) => {
+      if (!isDragging) return;
+
+      const currentY = e.touches[0].clientY;
+      const diff = Math.max(0, currentY - startYRef.current);
+      setDragY(diff);
+    },
+    [isDragging],
+  );
+
+  const handleTouchEnd = useCallback(() => {
+    handleDragEnd();
+  }, [handleDragEnd]);
+
+  useEffect(() => {
+    if (!isDragging) return;
+
+    const handleMouseMove = (e: MouseEvent) => {
+      const diff = Math.max(0, e.clientY - startYRef.current);
+      setDragY(diff);
+    };
+
+    const handleMouseUp = () => {
+      handleDragEnd();
+    };
+
+    document.addEventListener('mousemove', handleMouseMove);
+    document.addEventListener('mouseup', handleMouseUp);
+
+    return () => {
+      document.removeEventListener('mousemove', handleMouseMove);
+      document.removeEventListener('mouseup', handleMouseUp);
+    };
+  }, [isDragging, handleDragEnd]);
+
+  if (!isVisible) return null;
+
+  const sheetHeight = typeof height === 'number' ? `${height}px` : height;
+
+  return createPortal(
+    <div
+      ref={backdropRef}
+      className={`fixed inset-0 z-50 backdrop-blur-sm transition-all duration-300 ${
+        isOpen ? 'bg-neutral-100/50' : 'bg-transparent'
+      }`}
+      onClick={handleBackdropClick}
+    >
+      <div
+        ref={sheetRef}
+        className={`fixed right-0 bottom-0 left-0 mx-0 flex w-full max-w-none flex-col rounded-t-3xl bg-white shadow-2xl transition-transform duration-300 ease-out ${isOpen ? 'translate-y-0' : 'translate-y-full'} ${isDragging ? '' : 'transition-transform duration-300 ease-out'} ${className} `}
```

Comment:
```suggestion
        className={`fixed right-0 bottom-0 left-0 mx-0 flex w-full max-w-none flex-col rounded-t-3xl bg-white shadow-2xl ${isOpen ? 'translate-y-0' : 'translate-y-full'} ${isDragging ? '' : 'transition-transform duration-300 ease-out'} ${className} `}
```

의도하신 건 앞을 날리면 되는 부분일까요??